src/services/users.ts

```diff
@@ -54,9 +65,86 @@ export function getUserAttributes(dbPool: PostgresJsDatabase<typeof db>) {
   };
 }
 
+/**
+ * Checks user data for existing entries in the database.
+ * @param {PostgresJsDatabase<typeof db>} dbPool - The database connection pool.
+ * @param {string} userId - The ID of the user to check.
+ * @param {UserData} userData - The user data to check.
+ * @returns {Promise<Array<string> | null>} - An array of errors if user data conflicts, otherwise null.
+ */
+async function checkUserData(
```

Comment:
i think a rename here to `validateUserData` can give us a better overview on what is going on inside the function